### PR TITLE
Add SQL routine call for submissions

### PR DESCRIPTION
**Pull request by mosabua**

### CODE REVIEW

File: _posts/2023-11-07-routines.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+documentation]({{site_url}}/docs/current/routines/examples.html), and send you
+some goodies as a reward!
+
+<!--more-->
+
+With the recent [Trino 431
+release]({{site_url}}/docs/current/release/release-431.html) we shipped a
+feature that has been awaited by many Trino users for a long, long time. [SQL
+routines]({{site_url}}/docs/current/routines.html) are an easy way to define our
+own procedural, custom function. All users of your Trino instance can then use
```
Comment (colebow) [inline _posts/2023-11-07-routines.md:21]:
```suggestion
own procedural, custom functions. All users of your Trino instance can then use
```

Comment (colebow) [inline _posts/2023-11-07-routines.md:21]: Maybe also...
```suggestion
own procedural, custom function. All users on your Trino instance can then use
```

File: _posts/2023-11-07-routines.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+that function in their queries and enjoy the new feature to simplify their
+queries.
+
+The new process of simply writing a routine in your client tool in SQL, replaces
```
Comment (colebow) [inline _posts/2023-11-07-routines.md:25]:
```suggestion
The new process of writing a routine in your client tool in SQL replaces
```

File: _posts/2023-11-07-routines.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+the old way of having to create a custom plugin in Java, compiling it, and
+getting the binary deployed in your cluster. Time to using a function went from
```
Comment (colebow) [inline _posts/2023-11-07-routines.md:27]:
```suggestion
getting the binary deployed in your cluster. The time it takes to use a custom function has gone from
```

File: _posts/2023-11-07-routines.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+* `REPEAT`
+* `RETURN`
+* `SET`
+* `WHILE`
+
+With the memory connector and the Hive connector supporting routine storage, you
+can use `CREATE FUNCTION` and `DROP FUNCTION`, so that everyone using the
+cluster has access to your routines.
+
+The unit tests and our examples documentation contain a number of routines that
```
Comment (colebow) [inline _posts/2023-11-07-routines.md:49]:
```suggestion
The unit tests and our example documentation contain a number of routines that
```
Link to the documentation, and maybe even unit tests on GitHub?

File: _posts/2023-11-07-routines.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+scratch the surface of what is possible. Now we are looking for you to help us
```
Comment (colebow) [inline _posts/2023-11-07-routines.md:50]:
```suggestion
scratch the surface of what is possible. Now, we are looking for you to help us
```

File: _posts/2023-11-07-routines.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+improve the documentation and maybe even find some bugs. So here is what we are
+asking from you:
+
+* Send us your best SQL routine.
+  * Create a pull request to the [examples
```
Comment (colebow) [inline _posts/2023-11-07-routines.md:55]:
```suggestion
  * Create a pull request to add to the [example
```

File: _posts/2023-11-07-routines.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    documentation](https://github.com/trinodb/trino/blob/master/docs/src/main/sphinx/routines/examples.md)
+    that adds a new section and mention [Manfred (mosabua)](https://github.com/mosabua)
```
Comment (colebow) [inline _posts/2023-11-07-routines.md:57]:
```suggestion
    with a new section, and request a review from [Manfred (mosabua)](https://github.com/mosabua)
```

File: _posts/2023-11-07-routines.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  invocation.
+* Add any necessary tables or data so we can test the function.
+
+We plan to present the ten best submissions at [Trino Summit 2023]({% post_url
+2023-09-14-trino-summit-2023-announcement %}), write a blog post, add them to
+the documentation, and [Starburst](https://www.starburst.io/) will send you a
+cool reward.
+
+Also, if you have more great Trino usage to talk about and share, we would love
+to see your speaker proposal for Trino Summit.
```
Comment (colebow) [inline _posts/2023-11-07-routines.md:71]: Link?

**Review by colebow — CHANGES_REQUESTED** [review]

**Comment by colebow** [comment]
As a thought, I'm not confident we'll even get 10 submissions to show off at Trino Summit, so maybe we should be a little hesitant about promising that?

**Comment by mosabua** [comment]
> As a thought, I'm not confident we'll even get 10 submissions to show off at Trino Summit, so maybe we should be a little hesitant about promising that?

We can just talk about showing some .. let me rejig it all

File: _posts/2023-11-07-routines.md (hunk 4 above)
Comment (mosabua) [reply _posts/2023-11-07-routines.md:49]: Not the tests .. but sure.. the docs

File: _posts/2023-11-08-routines.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+With the recent [Trino 431
+release]({{site_url}}/docs/current/release/release-431.html) we shipped a
+feature that has been awaited by many Trino users for a long, long time. [SQL
+routines]({{site_url}}/docs/current/routines.html) are an easy way to define our
+own procedural, custom functions. All users on your Trino instance can then use
+that function in their queries and enjoy the new feature to simplify their
+queries.
+
+The new process of writing a routine in your client tool in SQL replaces the old
+way of having to create a custom plugin in Java, compiling it, and getting the
```
Comment (dain) [inline _posts/2023-11-08-routines.md:26]: We aren't replacing the old way.  The old was was really inaccessible to all but the most technical users.

File: _posts/2023-11-08-routines.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+binary deployed in your cluster. The time it takes to use a function has gone
+from hours to minutes!
```
Comment (dain) [inline _posts/2023-11-08-routines.md:28]:
```suggestion
binary deployed in your cluster. The time it takes to use a function has gone
from hours to a few commands!
```

File: _posts/2023-11-08-routines.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    documentation](https://github.com/trinodb/trino/blob/master/docs/src/main/sphinx/routines/examples.md)
+    with a new section, and request a review from [Manfred
+    (mosabua)](https://github.com/mosabua)
+  * Alternatively, [email the details](mailto:manfred@starburst.io) and submit a
+    [CLA](https://github.com/trinodb/cla) separately.
+* Explain the use case, what the routine does, and maybe also how it works.
+* Include the full statement for the `CREATE FUNCTION` definition and an example
+  invocation.
+* Add any necessary tables or data so we can test the function.
+
```
Comment (dain) [inline _posts/2023-11-08-routines.md:66]: Maybe add a sentence saying if you need help writing a function, ask on some channel on slack

File: _posts/2023-11-08-routines.md (hunk 9 above)
Comment (mosabua) [reply _posts/2023-11-08-routines.md:26]: Updated

File: _posts/2023-11-08-routines.md (hunk 11 above)
Comment (mosabua) [reply _posts/2023-11-08-routines.md:66]: Added

File: _posts/2023-11-09-routines.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title: "Share your best Trino SQL routine"
+author: "Martin Traverso, Dain Sunstrom, David Phillips, Manfred Moser"
```
Comment (martint) [inline _posts/2023-11-09-routines.md:4]: Typo: Sunstrom -> Sundstrom

File: _posts/2023-11-09-routines.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+We plan to present submissions at [Trino Summit 2023]({% post_url
+2023-09-14-trino-summit-2023-announcement %}), write a blog post, add them to
+the documentation, and [Starburst](https://www.starburst.io/) will send a cool
+reward for the ten best entries.
+
+Also, if you have more great Trino usage to talk about and share, we would love
+to see your [speaker proposal for Trino
+Summit](https://sessionize.com/trino-summit-2023/).
+
+We look forwards to see many great tricks from you all.
```
Comment (martint) [inline _posts/2023-11-09-routines.md:80]: "look forward to seeing"

Also, why "tricks"? Maybe "submissions", "creations" ?

File: _posts/2023-11-09-routines.md (hunk 12 above)
Comment (mosabua) [reply _posts/2023-11-09-routines.md:4]: Wow .. I thought I cut and paste that from elsewhere.. now had to check .. did not find any other offenders..

**Review by dain — APPROVED** [review]

**Review by martint — APPROVED** [review]

**Pull request merged**